### PR TITLE
Inlining and threading of regular broadcasting

### PR DESCRIPTION
**Pull request by simonbyrne**
Last fix to #1455.

Gets us to over 4 SYPD on a single GPU (https://buildkite.com/clima/climaatmos-target-gpu-simulations/builds/85#018ae2cc-8cef-4c77-a9d9-dc444f05645d)

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

File: src/DataLayouts/cuda.jl [hunk 1]
```diff
@@ -42,32 +42,16 @@ Base.similar(
...
-    j = cld(global_idx, nx)
-    h = ((global_idx-1) % (nx*nx)) + 1
-    =#
-
     i = CUDA.threadIdx().x
     j = CUDA.threadIdx().y
 
     h = CUDA.blockIdx().x
-    v = CUDA.blockIdx().y
+    v = CUDA.blockDim().y * (CUDA.blockIdx().y - 1) + CUDA.threadIdx().z
```
Comment (sriharshakandala) [inline src/DataLayouts/cuda.jl:49]: Should this be 
```
v = CUDA.blockDim().z * (CUDA.blockIdx().y - 1) + CUDA.threadIdx().z
```

File: src/DataLayouts/cuda.jl [hunk 2]
```diff
@@ -76,11 +60,12 @@ function knl_fill!(dest, val)
...
```
Comment (sriharshakandala) [inline src/DataLayouts/cuda.jl:63]: Should this be 
```
v = CUDA.blockDim().z * (CUDA.blockIdx().y - 1) + CUDA.threadIdx().z
```

Comment (simonbyrne) [reply src/DataLayouts/cuda.jl:63]: good catch, thanks

Comment (simonbyrne) [reply src/DataLayouts/cuda.jl:63]: i'm worried we didn't pick this up in the tests though...

**Review by sriharshakandala — APPROVED** [review]

File: src/DataLayouts/cuda.jl [hunk 3]
```diff
@@ -127,7 +115,10 @@ function Base.fill!(
 ) where {S, Nij, A <: CUDA.CuArray}
     _, _, _, Nv, Nh = size(dest)
     if Nv > 0 && Nh > 0
-        CUDA.@cuda threads = (Nij, Nij) blocks = (Nh, Nv) knl_fill!(dest, val)
+        Nv_per_block = fld(256, Nij * Nij)
```
Comment (sriharshakandala) [inline src/DataLayouts/cuda.jl:118]: Nv_per_block = min(Nv, fld(256, Nij * Nij))

File: src/DataLayouts/cuda.jl [hunk 4]
```diff
@@ -117,7 +102,10 @@ function Base.copyto!(
 ) where {S, Nij, A <: CUDA.CuArray}
     _, _, _, Nv, Nh = size(bc)
     if Nv > 0 && Nh > 0
-        CUDA.@cuda threads = (Nij, Nij) blocks = (Nh, Nv) knl_copyto!(dest, bc)
+        Nv_per_block = fld(256, Nij * Nij)
```
Comment (sriharshakandala) [inline src/DataLayouts/cuda.jl:105]: Nv_per_block = min(Nv, fld(256, Nij * Nij))

**Pull request merged**